controller/src/main/java/io/pravega/controller/metrics/ZookeeperConnectionMetrics.java

```diff
@@ -0,0 +1,26 @@
+/**
+ * Copyright (c) 2017 Dell Inc., or its subsidiaries. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ */
+package io.pravega.controller.metrics;
+
+import static io.pravega.shared.MetricsNames.CONTROLLER_ZK_SESSION_EXPIRATION;
+import static io.pravega.shared.MetricsNames.globalMetricName;
+
+/**
+ * Class to encapsulate the logic to report Controller metrics related to Zookeeper.
+ */
+public class ZookeeperConnectionMetrics extends AbstractControllerMetrics {
+
+    /**
+     * This method reports a new session expiration event in a Controller instance.
+     */
+    public void reportZKSessionExpiration() {
+        DYNAMIC_LOGGER.incCounterValue(globalMetricName(CONTROLLER_ZK_SESSION_EXPIRATION), 1);
```

Comment:
Does it match the dimensional way of doing metrics that we are proposing in a concurrent PR?

Reply:
Well, this is just a counter and we just export it in the same way as all the other counters in the Controller (e.g., Streams, Transactions). When the dimensions (i.e., tags) come into play, then we may need to add them to all the Controller metrics, right?

Reply:
This is the global metric so there is no additional dimensions applied (besides common tags such as host).